docs/installation/webserver/nginx.md

```diff
@@ -6,106 +6,263 @@ Nginx: Configuring the virtual host
+a host.
+
+```nginx
+# Redirect requests for */index.php to the same route minus the "index.php"
+if ($request_uri ~* "^(.*/)index\.php(?>/|)(.*)$") {
+    return                     301 $1$2;
+}
+
+# Default prefix match, as all URIs begin with /
+location / {
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Bolt backend access
+#
+# NOTE: If you set a custom branding path, you will need to change '/bolt/'
+#       here to match
+location ~* /bolt/(.*)$ {
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Generated thumbnail images
+location ~* /thumbs/(.*)$ {
+    access_log                 off;
+    log_not_found              off;
+    expires                    max;
+    add_header                 Pragma public;
+    add_header                 Cache-Control "public, mustrevalidate, proxy-revalidate";
+
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Don't log, and do cache, asset files
+location ~* ^.+\.(?:atom|bmp|bz2|css|doc|eot|exe|gif|gz|ico|jpe?g|jpeg|jpg|js|map|mid|midi|mp4|ogg|ogv|otf|png|ppt|rar|rtf|svg|svgz|tar|tgz|ttf|wav|woff|woff2|xls|zip)$ {
+    access_log                 off;
+    log_not_found              off;
+    expires                    max;
+    add_header                 Pragma public;
+    add_header                 Cache-Control "public, mustrevalidate, proxy-revalidate";
+}
+
+# Don't create logs for favicon.ico, robots.txt, or rss.jpg requests
+location = /(?:favicon.ico|robots.txt|rss.jpg) {
+    log_not_found              off;
+    access_log                 off;
+}
 ```
-fastcgi_pass unix:/path/to/php-fpm/socket.sock;
+
+
+#### `restrictions.conf`
+
+The `restrictions.conf` file defines a common set of restrictions for a Bolt
+site.
+
+```nginx
+# Block access to "hidden" files, i.e. file names that begin with a dot
+location ~ /\. {
+    deny                      all;
+}
+
+# Block PHP files from being run in upload (files), app, theme and extension directories
+location ~* /(?:extensions|files|theme)/(.*)\.php$ {
+    deny                      all;
+}
+
+# Block access to Sqlite database files, Apache .htaccess & .htpasswd files
+location ~ /\.(?:db|htaccess|htpasswd)$ {
+    deny                      all;
+}
+
+# Block access to the app, cache & vendor directories
+location ~ /(?:app|src|tests|vendor)/ {
+    deny                      all;
+}
+
+# Block access to Markdown, Twig & YAML files directly
+location ~* /(.*)\.(?:markdown|md|twig|yaml|yml)$ {
+    deny                      all;
+}
+
+# Block access to the app, cache & vendor directories
+#
+# NOTE: If you have one or more of 'app' , 'src', 'test' and 'vendor' as
+# sub-directories of your installation root, you should uncomment
+# this location block to prevent site visitors having access to the
+# various directories that contain executable code.
+#
+# NOTE: This approach is *not* recommended for production use.
```

Comment:
Yeah, but my point is that if we are going to have a comment saying "this is bad" we should have a link to what the proper solution is.